tests/api/mam_test.c

```diff
@@ -8,36 +8,183 @@
 
 #include <time.h>
 #include "accelerator/core/apis.h"
-#include "accelerator/core/proxy_apis.h"
-#include "test_define.h"
+#include "accelerator/core/mam_core.h"
+#include "common.h"
+#include "tests/test_define.h"
 
 static ta_core_t ta_core;
 struct timespec start_time, end_time;
 
 char driver_tag_msg[NUM_TRYTES_TAG];
-ta_send_mam_res_t* res;
+ta_send_mam_res_t res;
 
 #if defined(ENABLE_STAT)
 #define TEST_COUNT 100
 #else
 #define TEST_COUNT 1
 #endif
 
-void test_receive_mam_message(void) {
-  const char* json =
-      "{\"data_id\":{\"chid\":\"SYZJOQCGWGOTGGZUCZYQGHWWHQVIVBHJFHNXYXZQSGJXIHL9KMXNOBULIPTPGMBJSMFNBSAGCVVZ9MDMX\"},"
-      "\"protocol\":\"MAM_V1\"}";
+static void gen_rand_seed(char* trytes) {
+  const char tryte_alphabet[] = "NOPQRSTUVWXYZ9ABCDEFGHIJKLM";
+
+  for (int i = 0; i < NUM_TRYTES_HASH; i++) {
+    trytes[i] = tryte_alphabet[rand() % 27];
```

Comment:
Can we reuse the `gen_rand_trytes()` of https://github.com/DLTcollab/tangle-accelerator/pull/587 in `gen_rand_seed()`?
Such as `gen_rand_trytes(NUM_TRYTES_HASH, trytes)`.

Reply:
sure but I should wait for that PR to be merged

Reply:
Add FIXME in comments to address this proposed change.